File: venus_mobile/lib/models/booking_table.dart
class BookingTable {
  final String? id;
  final String name;
  final String email;
  final String phone_number;
  final String date;
  final String start_time;
  final String end_time;
  final int person_number;
  int? tableId; // Remove final

  BookingTable({
    this.id,
    required this.name,
    required this.email,
    required this.phone_number,
    required this.date,
    required this.start_time,
    required this.end_time,
    required this.person_number,
    this.tableId,
  });

  factory BookingTable.fromJson(Map<String, dynamic> json) {
    return BookingTable(
      id: json['id'] as String?,
      name: json['name'] as String,
      email: json['email'] as String,
      phone_number: json['phone_number'] as String,
      date: json['date'] as String,
      start_time: json['start_time'] as String,
      end_time: json['end_time'] as String,
      person_number: json['person_number'] as int,
      tableId: json['tableId'] as int?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'email': email,
      'phone_number': phone_number,
      'date': date,
      'start_time': start_time,
      'end_time': end_time,
      'person_number': person_number,
      'tableId': tableId,
    };
  }

  @override
  String toString() {
    return 'BookingTable{id: $id, name: $name, email: $email, phone_number: $phone_number, date: $date, start_time: $start_time, end_time: $end_time, person_number: $person_number, tableId: $tableId}';
  }
}

File: RestaurantService/src/main/java/com/restaurant/service/entities/BookingTable.java
package com.restaurant.service.entities;

import java.text.SimpleDateFormat;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import jakarta.persistence.Transient;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@AllArgsConstructor
@EqualsAndHashCode(callSuper = false)
@Entity
@Table(name = "bookingTable")
public class BookingTable extends AbstractEntity {
	private static final long serialVersionUID = 1L;

	@NotBlank(message = "Name is required")
	@Column(name = "name")
	private String name;

	@Email(message = "Email must be valid")
	@NotBlank(message = "Email is required")
	@Column(name = "email")
	private String email;

	@Pattern(regexp = "^0\\d{9,10}$", message = "Phone number must start with 0 and have 9-10 digits")
	@NotBlank(message = "Phone number is required")
	@Column(name = "phone_number")
	private String phone_number;

	@NotBlank(message = "Date is required")
	@Column(name = "date")
	private String date;

	@NotBlank(message = "Start time is required")
	@Column(name = "start_time")
	private String start_time;

	@NotBlank(message = "End time time is required")
	@Column(name = "end_time")
	private String end_time;

	@NotNull(message = "Number of persons is required")
	@Column(name = "person_number")
	private Integer person_number;

	@Column(name = "status")
	private Integer status = 0;

	@Column(name = "special")
	private Integer special = 0;

	@Column(name = "tableId")
	private Long tableId;

//	@ManyToOne
//	@JoinColumn(name = "tableId", referencedColumnName = "id")
//	private RestaurantTable restaurantTable;

	public BookingTable() {
		super();
	}

	public BookingTable(String name, String email, String phone_number, String date, String start_time, String end_time,
			int person_number) {
		super();
		this.name = name;
		this.email = email;
		this.phone_number = phone_number;
		this.date = date;
		this.start_time = start_time;
		this.end_time = end_time;
		this.person_number = person_number;
	}

//	@Transient
//	public String getDateFormat() {
//		SimpleDateFormat formatter = new SimpleDateFormat("dd-MM-yyyy");
//		return formatter.format(date);
//	}

	public BookingTable(String name, String email, String phone_number, String date, String start_time, String end_time,
			Integer person_number, Long tableId) {
		super();
		this.name = name;
		this.email = email;
		this.phone_number = phone_number;
		this.date = date;
		this.start_time = start_time;
		this.end_time = end_time;
		this.person_number = person_number;
		this.tableId = tableId;
	}

}
